#! /bin/sh -e

# All lines beginning with `# DPATCH:' are a description of the patch.
# DP: don't swap operands during reloads

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p1 < $0
        #cd ${dir}gcc && autoconf
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
        #rm ${dir}gcc/configure
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

---

 gcc/recog.c |    1 +
 1 file changed, 1 insertion(+)

Index: gcc-4.2/gcc/recog.c
===================================================================
--- gcc-4.2.orig/gcc/recog.c
+++ gcc-4.2/gcc/recog.c
@@ -531,6 +531,7 @@ validate_replace_rtx_1 (rtx *loc, rtx fr
      simplifications, as it is not our job.  */
 
   if (SWAPPABLE_OPERANDS_P (x)
+      && !reload_in_progress
       && swap_commutative_operands_p (XEXP (x, 0), XEXP (x, 1)))
     {
       validate_change (object, loc,
